x86,amd_detect_cmp: Dont forget to retrieve cpu number

In non-NUMA case we have to retrieve cpu number in a sake
of HT compatibility.

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
---
 arch/x86_64/kernel/setup.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.18-ovz/arch/x86_64/kernel/setup.c
=====================================================================
--- linux-2.6.18-ovz.orig/arch/x86_64/kernel/setup.c
+++ linux-2.6.18-ovz/arch/x86_64/kernel/setup.c
@@ -857,8 +857,8 @@ static void __init amd_detect_cmp(struct
 {
 #ifdef CONFIG_SMP
 	unsigned bits;
-#ifdef CONFIG_NUMA
 	int cpu = smp_processor_id();
+#ifdef CONFIG_NUMA
 	int node = 0;
 	unsigned apicid = hard_smp_processor_id();
 #endif
